File: app/src/main/java/com/jintin/app/MainActivity.java
package com.jintin.app;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ListView listView = findViewById(R.id.listView);
        String[] items = new String[]{"Basic", "Grid", "Multi-Holder", "Dynamic"};
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, items);
        listView.setAdapter(adapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                switch (position) {
                    case 0:
                        startActivity(new Intent(MainActivity.this, BasicActivity.class));
                        break;
                    case 1:
                        startActivity(new Intent(MainActivity.this, GridActivity.class));
                        break;
                    case 2:
                        startActivity(new Intent(MainActivity.this, MultiTypeActivity.class));
                        break;
                    case 3:
                        startActivity(new Intent(MainActivity.this, DynamicActivity.class));
                        break;
                }
            }
        });
    }
}

File: app/src/main/java/com/jintin/app/MainActivity.kt
package com.jintin.app

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter
import android.widget.ListView
import kotlin.reflect.KClass

/**
 * Project root
 */
class MainActivity : AppCompatActivity() {

    /**
     * Item to show in List
     */
    class ListItem(val title: String, val kClass: KClass<out Activity>) {
        override fun toString(): String {
            return title
        }
    }

    private val items = arrayOf(
            ListItem("Basic", BasicActivity::class),
            ListItem("Grid", GridActivity::class),
            ListItem("Multi-Holder", MultiTypeActivity::class),
            ListItem("Dynamic", DynamicActivity::class)
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val listView = findViewById<ListView>(R.id.listView)
        listView.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, items)
        listView.setOnItemClickListener { _, _, position, _ -> showActivity(position) }
    }

    private fun showActivity(position: Int) {
        startActivity(Intent(this, items[position].kClass.java))
    }
}
